flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -228,7 +234,14 @@ public void startNewWorker(ResourceProfile resourceProfile) {
 
 	@Override
 	public void stopWorker(ResourceID resourceID) {
```

Comment:
I actually think that we should refactor this method to take a parameter of generic type `WorkerType` which contains all the framework specific registration information. In the current `YarnResourceManager` implementation case, it would be `ResourceID` which is generated from the container id. Thus, we should be able to retrieve the `containerId` from it.

Reply:
Do you mean something like MesosWorkerStore.Worker in MesosResourceManager, which has both the taskId (used as resourceID) and other context info for the Mesos worker?

Reply:
Yes exactly.